### ISSUE TITLE
user_some + AND doesn't seem to be working properly

### ISSUE THREAD

**Issue by Mercurial** [prisma/prisma#3943]
![image](https://user-images.githubusercontent.com/585569/51848974-ab07fe00-2359-11e9-936a-85d45ab9f409.png)

As the title says this query doesn't seem to be working properly

this is my data

![image](https://user-images.githubusercontent.com/585569/51849033-cf63da80-2359-11e9-8872-80bef52a1006.png)

![image](https://user-images.githubusercontent.com/585569/51849065-e0ace700-2359-11e9-8a4f-0ce37b3390bd.png)

basically I'm trying to get all connections that contains these two user ids in the "users" field

**Comment by pantharshit00** [comment]
cc @do4gr 

I think this is related to #3897

**Comment by do4gr** [comment]
Hey, thanks for your question. 

The semantic of our _some relational filter is a bit different from what you are trying to achieve. The nested condition blog below the _some (every and none are the same) is checked in complete against every node. So your condition is always false in our semantic. 

You are trying to express: Connections that have a user with id1 AND another user with id2.
To us the query asks: Connections that have a user with id1 AND that same user also has id2. 

You can rewrite your query like this to express what you want:

```
connections(
  AND:[
  {users_some:{id: "firstid"}},
  {users_some:{id: "secondid"}}
  ]
)
{
  id,
  status
}
```
